homeassistant/components/modbus/climate.py

```diff
@@ -387,6 +451,22 @@ async def async_update(self, now: datetime | None = None) -> None:
                     int(fan_mode), self._attr_fan_mode
                 )
 
+        # Read the Swing mode register if defined
+        if self._swing_mode_register is not None:
+            swing_mode = await self._async_read_register(
+                CALL_TYPE_REGISTER_HOLDING,
+                self._swing_mode_register
+                if isinstance(self._swing_mode_register, int)
+                else self._swing_mode_register[0],
+                raw=True,
+            )
+
+            # Translate the value received
+            if swing_mode is not None:
```

Comment:
This looks wrong, if the swing_mode register do not respond, the swing_mode should be set to unknown or something similar, so the user sees the problem.

A log entry would also be good, since it might be a bad configuration.

Reply:
Tried to fix it: https://github.com/home-assistant/core/pull/113710/commits/3df9758b9ac2d6e6383971faafcc6445c01903de   I removed the test against None because it seems that in case of missing answer, the bus answers -1. and it is never None. 

If you like this way, later (other PR) I could apply the same also to fan_mode and hvac_mode.